src/util/PriorityQueue.hxx

```diff
@@ -0,0 +1,62 @@
+#ifndef PRIORITY_QUEUE_HXX_
+#define PRIORITY_QUEUE_HXX_
+
+#include <vector>
+
```

Comment:
I'm not sure that providing `::begin` and `::end` is a good idea... In a min/max heap, there is no clear order of how you should iterate. I understand that you need it for `::remove` but I think that in that case, the interface you provide is forcing a complexity of n log n

Reply:
I think a better alternative would be to create a function like this that would take care of erasing everything that you want. This function is also O(n)
```suggestion
template <typename Eraser>
void erase_if(Eraser && eraser = {}) {
  m_container.erase(std::remove_if(m_container.begin(), m_container.end(), eraser), m_container.end());
  std::make_heap(m_container.begin(), m_container.end(), Comparator());
}
```

Reply:
Done